### PR TITLE
Fix PHP 8.2 deprecation warning

### LINKED ISSUE #268: PHP 8.2 Deprecated: Creation of dynamic property in Laminas\Db\Adapter\Driver\Pdo\Pdo->setProfiler()
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 2.16.1

#### Summary

setProfiler() method of Laminas\Db\Adapter\Driver\Pdo\Pdo dynamically creates $profiler property, but this behaviour is deprecated in PHP 8.2:
```
PHP Deprecated:  Creation of dynamic property Laminas\Db\Adapter\Driver\Pdo\Pdo::$profiler is deprecated in .../laminas/laminas-db/src/Adapter/Driver/Pdo/Pdo.php on line 73
```

#### Current behavior

Deprecation warning

#### How to reproduce

Using PHP 8.2:

```php
ini_set('error_reporting', E_ALL);
$driver = new Laminas\Db\Adapter\Driver\Pdo\Pdo([]);
$driver->setProfiler(new Laminas\Db\Adapter\Profiler\Profiler());
```

#### Expected behavior

No deprecation warning.

### PR DESCRIPTION
**Pull request by Grundik**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

Fixes #268, improving PHP 8.2 compatibility.

### CODE REVIEW

File: src/Adapter/Driver/Pdo/Pdo.php [hunk 1]
```diff
@@ -37,6 +37,9 @@ class Pdo implements DriverInterface, DriverFeatureInterface, Profiler\ProfilerA
     /** @var array */
     protected $features = [];
 
+    /** @var Profiler\ProfilerInterface */
+    protected $profiler;
```
Comment (samsonasik) [inline src/Adapter/Driver/Pdo/Pdo.php]: need to have `public` modifier, unfortunatelly, ref https://3v4l.org/HYEML

Comment (Grundik) [reply src/Adapter/Driver/Pdo/Pdo.php]: There are public getProfiler() method, but sure, lets not introduce BC break.

File: src/Adapter/Driver/Pdo/Pdo.php [hunk 2]
```diff
@@ -37,6 +37,9 @@ class Pdo implements DriverInterface, DriverFeatureInterface, Profiler\ProfilerA
     /** @var array */
     protected $features = [];
 
+    /** @var Profiler\ProfilerInterface */
```
Comment (Ocramius) [inline src/Adapter/Driver/Pdo/Pdo.php]: Let's mark it `@internal` to prevent abuse?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Adapter/Driver/Pdo/Pdo.php (hunk 2 above)
Comment (Grundik) [reply src/Adapter/Driver/Pdo/Pdo.php]: Done

File: src/Adapter/Driver/Pdo/Pdo.php [hunk 3]
```diff
@@ -37,6 +37,12 @@ class Pdo implements DriverInterface, DriverFeatureInterface, Profiler\ProfilerA
     /** @var array */
     protected $features = [];
 
+    /**
+     * @var Profiler\ProfilerInterface
```
Comment (Ocramius) [inline src/Adapter/Driver/Pdo/Pdo.php]: ok, one last bit (sorry)! 

run `vendor/bin/phpcbf` on this and it should fix the CS issue (probably just sprtijg issue)

Comment (Grundik) [reply src/Adapter/Driver/Pdo/Pdo.php]: Done

**Comment by Ocramius** [comment]
Failures in latest dependencies are unrelated to this patch, and caused by upstream PHPUnit deprecations.

**Review by Ocramius — APPROVED** [review]
Thanks @Grundik!

**Pull request merged**